# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/manager/ssl/TransportSecurityInfo.h
# Commit: c770a884ed23
# Full Hash: c770a884ed23f610ab0905785c173906444f0382
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-16 05:20:00
# Regressor Bug: 1580315
# File Overlap Count: 1
# Description:
#   Bug 1580315 - Change certList internal representation to Array r=keeler
#   
#   This patch intends to change the internal reprensentation of certList
#   from nsIX509CertList to Array for TransportSecurityInfo.
#   
# ==============================================================================

diff -r 1f7358c2efa5 -r c770a884ed23 security/manager/ssl/TransportSecurityInfo.h
--- a/security/manager/ssl/TransportSecurityInfo.h	Wed Oct 16 00:34:01 2019 +0300
+++ b/security/manager/ssl/TransportSecurityInfo.h	Tue Oct 15 19:57:23 2019 +0000
@@ -121,7 +121,11 @@
 
  protected:
   nsCOMPtr<nsIInterfaceRequestor> mCallbacks;
-  nsCOMPtr<nsIX509CertList> mSucceededCertChain;
+  nsTArray<RefPtr<nsIX509Cert>> mSucceededCertChain;
+
+  static nsresult ConvertCertArrayToCertList(
+      const nsTArray<RefPtr<nsIX509Cert>>& aCertArray,
+      nsIX509CertList** aCertList);
 
  private:
   uint32_t mSecurityState;
@@ -135,9 +139,12 @@
   nsCOMPtr<nsIX509Cert> mServerCert;
 
   /* Peer cert chain for failed connections (for error reporting) */
-  nsCOMPtr<nsIX509CertList> mFailedCertChain;
+  nsTArray<RefPtr<nsIX509Cert>> mFailedCertChain;
 
   nsresult ReadSSLStatus(nsIObjectInputStream* aStream);
+  static nsresult ConvertCertListToCertArray(
+      const nsCOMPtr<nsIX509CertList>& aCertList,
+      nsTArray<RefPtr<nsIX509Cert>>& aCertArray);
 };
 
 class RememberCertErrorsTable {
